### PR TITLE
Integrate module to AMD babel plugin to jsTransform.

### PR DESCRIPTION
**Pull request by aomarks**
Also:
- Add unit tests to `jsTransform`.
- Expose `isComponentRequest` option for Polyserve to use.
- Renamed `compile` to `compileToEs5` to clarify what it does.

Part of #269.

(Note this does not yet include the dependency chaining from https://github.com/Polymer/polyserve/commit/fc33cd50bfa87bf187be797ff7c336dece05c95c which I intend to add in a follow up).

 - [x] CHANGELOG.md has been updated

### CODE REVIEW

File: src/test/js-transform_test.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+      import { dep1 } from 'dep1';
+      import { dep2 } from 'dep2';
+      import { dep2A } from 'dep2/a';
+      import { dep3 } from 'dep3';
+      import { dep4 } from 'dep4';
+
+      import { p1 } from '/already/a/path.js';
+      import { p2 } from './already/a/path.js';
+      import { p3 } from '../already/a/path.js';
+      import { p4 } from '../already/a/path.js';
```
Comment (justinfagnani) [inline src/test/js-transform_test.ts]: Why the duplication here? What's it testing?

Comment (aomarks) [reply src/test/js-transform_test.ts]: Since the refactoring, I wanted to associate tests for jsTransform features directly in a jsTransform test, instead of via optimize-streams, because this function will now be shared with Polyserve and stands alone. I removed some edge cases from the optimize-stream test, since now we just care that the transformation ran at all.
